### PR TITLE
[ JPG - 009] Update Transaction

### PR DESCRIPTION
**Pull request by Bhavesh-SP18**

### CODE REVIEW

**Comment by Ketan-SP18** [comment]
Add Test cases

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 1]
```diff
@@ -26,6 +28,7 @@ public TransactionService(TransactionRepository transactionRepository, UserRepos
 
     /**
      * Create method to do the actual transaction contaning the business logic to add deposit and withdraw and return response according
+     *
```
[the remark this answers is not in the document]
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:31]: I used a shortcut to generate a java doc. mistakenly added space while writing a java doc of update transaction.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 2]
```diff
@@ -45,6 +48,7 @@ public TransactionResponse transact(Integer userID, TransactionRequest request)
     /**
      * This method(getZeroOrLastBalance) is to get balance as zero if its user first transaction or the closing balance of the latest transaction.
      * first assign the balance 0.
+     *
```
[the remark this answers is not in the document]
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:51]: the same thing happened here.

**Review by Aashishverma06 — APPROVED** [review]
Approved.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 3]
```diff
@@ -22,81 +21,27 @@ public class TransactionService {
...
-                        .time(LocalDateTime.now())
-                        .transactionType(request.getTransactionType())
-                        .build());
-                break;
-            default:
-                return TransactionResponse.builder()
-                        .message(String.format("Unhandled transaction type %s", request.getTransactionType()))
-                        .build();
-        }
+        // TODO: Add deposit/withdraw logic and change below return accordingly
```
Comment (Hrishikesh-SP18) [inline src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:42]: is this required here.

File: src/main/java/com/passbook/sparkeighteen/service/TransactionService.java [hunk 4]
```diff
@@ -107,25 +52,41 @@ public TransactionResponse transact(Integer userID, TransactionRequest request)
      */
     private Float getUpdatedBalance(UserEntity user) {
         Float balance = 0f;
-        Optional<List<TransactionEntity>> optionalTxns = transactionRepository.findByUser(user);
-        if (optionalTxns.isPresent()) {
-            List<TransactionEntity> txns = optionalTxns.get();
-            if (txns.size() > 0) balance = txns.get(txns.size() - 1).getClosingBalance();
-        }
+        // TODO: Add getting the latest closing balance of the user
```
Comment (Hrishikesh-SP18) [inline src/main/java/com/passbook/sparkeighteen/service/TransactionService.java:55]: same here

**Pull request closed without merging**